roles/jboss_ws/tasks/main.yml

```diff
@@ -0,0 +1,93 @@
+  ignore_errors: yes
+
+- name: set jws_home based on find_jws_home
+  set_fact:
+      jws_home: "{{ find_jws_home['results'][0]['stdout_lines'][0] }}" 
+  ignore_errors: yes
+
+# ------------------------------ DETECT JWS/EWS -------------------------------
+- name: check jws presence with yum
+  raw: yum grouplist jws3 jws3plus jws5 | grep -A1 'Installed Groups:' | grep 'Red Hat JBoss Web Server [0-99]'
+  register: check_installed_with_rpm
+  ignore_errors: yes
+
+- name: determine if tomcat is installed from upstream or as part of redhat product
+  raw: \[ $(grep -iR redhat "{{ jws_home }}"/tomcat[0-99] 2>/dev/null| wc -l) != 0 \] && echo True || echo False
+  register: tomcat_is_not_upstream
+  ignore_errors: yes
+
+- name: set tomcat_is_part_of_redhat_product based on tomcat_is_not_upstream 
+  set_fact:
+      tomcat_is_part_of_redhat_product: "{{ tomcat_is_not_upstream['stdout_lines'][0] }}"
+  ignore_errors: yes
+
+- name: set installed_with_rpm based on check_installed_with_rpm 
+  set_fact:
+      installed_with_rpm: "{{ check_installed_with_rpm['stdout_lines']}}"
+  ignore_errors: yes
+
+  # detect jws if installed with zip even when no tomcat process is running
+- name: see if JBossEULA.txt exists 
+  raw: ls "{{ jws_home }}"/JBossEULA.txt
+  register: jboss_eula_txt_file
+  ignore_errors: yes
+
+- name: set jboss_eula_location based on jboss_eula_txt_file 
+  set_fact:
+      jboss_eula_location: "{{ jboss_eula_txt_file['stdout_lines'][0] }}"
+  ignore_errors: yes
+# Can't detect when all packages are installed with rpm, but not as a group...
+
+# ------------------------------ FIND JWS VERSION -----------------------------
+  # Additional methods of finding different versions can be appended to the list
+  # under 'with_items' to create array of possibilities for the version. A map of
+  # the raw stdout value to the correctly formatted version string should be appended
+  # to JWS_CLASSIFICATIONS in jboss_web_server.py
+
+  # 'with_items' raw commands purpose:
+  # Ver 2.0.1 and below have a unique combination of httpd and tomcat versions
+  # Ver 3.x.x mentioned in id tags in “TOMCAT_HOME/webapps/docs/changelog.html”
+  # Ver 5.0.0 has version.txt in JWS_HOME when installed with zip file
+  # Ver 5.x.x has a jws5-tomcat.service unit installed.
+
+- name: find JWS version
+  raw: "{{ item }}"
+  register: jws_version
+  ignore_errors: yes
+  with_items: 
+    - ."{{ jws_home }}"/httpd/sbin/httpd -v | grep Apache 2>/dev/null
+    - awk -F 'id=' '/JWS/ {print substr($2,16,9)}' "{{ jws_home }}"/tomcat?/webapps/docs/changelog.html | grep JWS | head -n1 2>/dev/null
+    - cat "{{ jws_home }}"/version.txt | grep "Red Hat JBoss Web Server" 2>/dev/null
+    - systemctl status jws5-tomcat.service | grep "Apache Tomcat Web" | grep -o jws5
+    - (echo -n $("{{ jws_home }}"/httpd/sbin/httpd -v | grep Apache); echo $(./"{{ jws_home }}"/tomcat*/bin/version.sh | grep "Server version")) | sed -e 's/Server version: //g'
```

Comment:
if you add the when conditions above you would need to add a check that `jws_home` is defined.
```
when: jws_home is defined
```

Looks like you could still to the `systemctl` check.

You should check that they have `systemctl` older systems wont they will have `chkconfig` with the `internal_have_systemctl` fact:
https://github.com/quipucords/quipucords/blob/master/roles/jboss_eap/tasks/main.yml#L160

I assume you should also add a check using chkconfig as well.
https://github.com/quipucords/quipucords/blob/master/roles/jboss_eap/tasks/main.yml#L167

Each of these items should have the `2>/dev/null` before the first pipe.